go.mod

```diff
@@ -1,6 +1,6 @@
 module go.etcd.io/bbolt
 
-go 1.21
+go 1.21.4
```

Comment:
Hey @arjunmalhotra1 - I don't think it is agreed to start pinning `go.mod` to a specific patch version. The community has only just started discussing it in https://github.com/etcd-io/etcd/issues/16828.

Please keep the `go.mod` at `1.21` for now.